go/mysql/fastparse/fastparse.go

```diff
@@ -139,6 +151,22 @@ next:
 			break next
 		}
 
+		var cutoff uint64
+		switch base {
+		case 10:
+			cutoff = math.MaxInt64/10 + 1
+		case 16:
+			cutoff = math.MaxInt64/16 + 1
+		default:
+			cutoff = math.MaxInt64/uint64(base) + 1
+		}
+		if d >= cutoff {
+			if minus {
+				return math.MinInt64, fmt.Errorf("cannot parse int64 from %q: %w", s, ErrOverflow)
+			}
+			return math.MaxInt64, fmt.Errorf("cannot parse int64 from %q: %w", s, ErrOverflow)
+		}
+
```

Comment:
seems duplicate code, can it be moved to a method?

Reply:
@harshit-gangal It's not exactly duplicate, it's slightly different with different types and different maximum values. Trying to extract it would then defeat things like the compiler optimization here for base 10 / 16.